service/shadowsocks.go

```diff
@@ -73,14 +73,25 @@ func NewShadowsocksService(opts ...Option) (Service, error) {
 
```

Comment:
I'm worried that at this point your logger may be nil, exposing the code to crashes. The constructor should always set it, so we have one single place to define what the default logger is. Otherwise each function may have its own logic to decide on the default logger.

We also need a comment specifying what the default logger is (noop? system default?).

Same for Packet.